meson.build

```diff
@@ -0,0 +1,176 @@
+    'format=2',
+
+    # Warn if the format string requires an unsigned argument and the argument is signed and vice versa
+    'format-signedness',
+
+    # Warn about anything that depends on the “size of” a function type or of void
+    'pointer-arith',
+
+    # Warn if a function is declared or defined without specifying the argument types
+    'strict-prototypes',
+
+    # Warn if a variable-length array is used
+    'vla',
+
+    # Give string constants the type const char[length] so that copying the address of one into a non-const char * pointer produces
+    # a warning
+    'write-strings',
+]
+
+foreach warning : warning_enable
+    if meson.get_compiler('c').has_argument('-W' + warning)
+        add_project_arguments('-W' + warning, language: 'c')
+    endif
+endforeach
+
+####################################################################################################################################
+# Disable various unhelpful warnings by adding -Wno-warning switches. GCC won't complain about unrecognized -Wno-warning switches,
+# so we have to test for the positive form, and if that works, add the negative form. Meson will automatically convert
+# -Wno-warning switches as of 0.51 but leave the check here in case we decide to use an earlier version since the code would just
+# silently stop working as expected.
+####################################################################################################################################
+warning_disable = [
+    # Warn for variables that might be changed by longjmp or vfork. Disable because of constant false positives/negatives.
+    'clobbered',
+
+    # Warn if a structure’s initializer has some fields missing. Disable so we can initialize with {0}.
+    'missing-field-initializers',
+
+    # Warn when a switch case falls through. Disable because this an useful aspect of switches and tests should catch problems.
+    'implicit-fallthrough',
+]
+
+foreach warning : warning_disable
+    if meson.get_compiler('c').has_argument('-W' + warning)
+        add_project_arguments('-Wno-' + warning, language: 'c')
+    endif
+endforeach
+
+####################################################################################################################################
+# Enable additional optimizations if the level is high enough
+####################################################################################################################################
+if get_option('optimization') in ['2', '3']
+    optimization_enable = [
+        # Unroll loops whose number of iterations can be determined at compile time or upon entry to the loop
+        'unroll-loops',
+
+        # Perform loop vectorization on trees
+        'tree-vectorize',
+    ]
+
+    foreach optimization : optimization_enable
+        if meson.get_compiler('c').has_argument('-f' + optimization)
+            add_project_arguments('-f' + optimization, language: 'c')
```

Comment:
Similarly here, you can list it with -f in the array and simplify the add_project_arguments call.

Actually you can also do something like:

```meson
add_project_arguments(cc.get_supported_arguments(warning_enable, optimization_enable), language: 'c')
```

Reply:
What some people do in this situation is check `if get_option('foo') in ['2', '3']` then e.g. `all_checked_options += [...]` and then test+add them all at the end.

Reply:
Hmmm, I can see that, but still prefer breaking things up a bit, as in ca61ddc93.

Reply:
Fair enough. I think keeping the lists separate has some documentary value. Once these build files are written they won't get touched much so clearer is better. At least clearer to me. :)